src/Backend/Pinterest.php

```diff
@@ -29,16 +31,16 @@ public function getRequest($url)
     /**
      * {@inheritdoc}
      */
-    public function filterResponse($content)
+    public function filterResponse(string $content): string
     {
-        return mb_substr($content, 2, mb_strlen($content) - 3);
+        return mb_substr($content, 2, -3);
```

Comment:
Funny, because this is actually not even documented for this function. https://www.php.net/manual/de/function.mb-substr.php

Reply:
But mentioned here https://www.php.net/manual/en/function.mb-substr.php (and have link to this function on mentioned page)

Reply:
I don't see any mention of a negative $length parameter. Neither on EN nor DE page. But does not matter, it works. ;-)

Reply:
Sorry, my fault. Correct link is https://www.php.net/manual/en/function.substr.php

Reply:
The third parameter would have to be -1 instead of -3, though, because you also have to take the offset (second parameter) into account!

Reply:
@daiwai according to [manual](https://www.php.net/manual/en/function.substr.php) changed line of code have the same meaning as before change
```
If length is given and is negative, then that many characters will be omitted from the end of string (after the start position has been calculated when a start is negative). If start denotes the position of this truncation or beyond, FALSE will be returned. 
```
Would you be so kind to add some context to your comment (with some examples)? I will write some tests for old code and then run the same tests against new.

Thank you in advance

Reply:
@vkhramtsov The intention in this line is to strip 2 characters from the front of the string and 1 from the end.

The old implementation used a positive value for $length but because the sub string doesn't start at offset 0 but at 2 of the string we need to also substract the offset (value of the $start parameter) from the total string length to get the right value for $length.

Let's say `$len` = the length of the string, `$s` = the number of characters to omit form the start of the string and `$e` is the number of characters we wish to omit from the end of the string, the calls to (mb_)substr have to look like:

`$stripped = substr($str, $s, $e * - 1)` if we want to use a negative value for $length
or
`$stripped = substr($str, $s, $len - $s - $e);` if we want to use a positive value for $length

Reply:
Got it

Reply:
Fixed